apps/chat/src/components/Chat/Publish/PublicationHandler/ReviewRowItems/PublicationToolsetRow.tsx

```diff
@@ -4,25 +4,37 @@ import { getFolderIdFromEntityId } from '@/src/utils/app/folders';
 
 import { BackendResourceTypeName, EntityType } from '@/src/types/common';
 
+import { useAppSelector } from '@/src/store/hooks';
+import { ToolsetSelectors } from '@/src/store/selectors';
+
 import { ModelIcon } from '@/src/components/Chatbar/ModelIcon';
 
 import { PublicationItemRow } from './PublicationItemRow';
+import { PublicationItemProps } from './view-props';
+
+export const PublicationToolsetRow: React.FC<PublicationItemProps> = ({
+  item,
+  level,
+}) => {
+  const toolsets = useAppSelector(ToolsetSelectors.selectToolsets);
+  const publishRequestToolsets = useAppSelector(
+    ToolsetSelectors.selectPublishRequestToolsets,
+  );
 
-import { ShareEntity } from '@epam/ai-dial-shared';
-
-interface Props {
-  item: ShareEntity;
-  level: number;
-}
+  const agent = useMemo(() => {
```

Comment:
I suggest to change naming to avoid confusion.
```suggestion
  const toolset = useMemo(() => {
```

Reply:
I've renamed it in the next PR, which should be merged right after